src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -107,6 +107,22 @@ export const walletSupportsChain = ({
   }
 }
 
+export const walletSupportsChain = ({
+  chainId,
+  chainAccountIds,
+  wallet,
+  isSnapInstalled,
+}: WalletSupportsChainArgs): boolean | null => {
+  // If the user has no connected chain account ids, the user can't use it to interact with the chain
+  if (!chainAccountIds.length) return false
```

Comment:
Doesn't look right - we now have a util and sub-util, both being consumed in different places and making implementation/consumption intent hard to understand.

Also see https://github.com/shapeshift/web/pull/7029/files#r1621989077, the new `walletDeviceSupportsChain` with its new runtime support checks without accountIds check seems off.

Would like to discuss this specific refactor some more to understand the intent and possibly find a better solution here

Reply:
Lets discuss - we're looking at this with different definitions of the intent of this util